tests/migration_test.go

```diff
@@ -3384,48 +3379,35 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 						cfg.MigrationConfiguration.NodeDrainTaintKey = &drain
 						tests.UpdateKubeVirtConfigValueAndWait(cfg)
 					}
-					setControlPlaneUnschedulable(true)
+
+					controlPlaneNodes, err := virtClient.CoreV1().Nodes().
+						List(context.Background(), metav1.ListOptions{LabelSelector: `node-role.kubernetes.io/control-plane`})
+					Expect(err).ShouldNot(HaveOccurred(), "could not list control-plane nodes")
+					Expect(controlPlaneNodes.Items).ShouldNot(BeEmpty(), "There are no control-plane nodes in the cluster")
+
+					// Make sure the vmi will not be scheduled in the control-plane node in those clusters where there is only one.
+					libnode.SetNodeUnschedulable(controlPlaneNodes.Items[0].Name, virtClient)
+					controlPlaneNodeSetUnschedulable = controlPlaneNodes.Items[0].Name
 				})
 
 				AfterEach(func() {
-					setControlPlaneUnschedulable(false)
+					resetNodeSchedulable()
```

Comment:
Calling `libnode.SetNodeSchedulable()` directly would be clearer IMO

Reply:
There was a debate within me about which one would be better. Now it's 2vs1! :) Thanks. Done